Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -1221,8 +1221,15 @@ private boolean readAndStripParam(Uri data, Activity activity) {
             try {
                 if (data != null) {
                     if (externalUriWhiteList_.size() > 0) {
-                        String externalIntentScheme = Uri.parse(data.toString()).getScheme();
-                        if (externalIntentScheme != null && externalUriWhiteList_.contains(externalIntentScheme)) {
+                        String uriString = data.toString();
+                        boolean foundWhiteListMatch = false;
+                        for (String whiteListScheme : externalUriWhiteList_) {
```

Comment:
It's not quite this simple, as we need to automatically whitelist "open" for default behavior, and I have concerns about treating whitelisted schemes and whitelisted paths the same.